scipy/_lib/deprecation.py

```diff
@@ -206,14 +214,26 @@ def _inner_deprecate_positional_args(f):
             elif param.kind == Parameter.KEYWORD_ONLY:
                 kwonly_args.append(name)
 
+        def warn_deprecated_args(kwargs):
+            intersection = deprecated_args.intersection(kwargs)
+            if intersection:
+                message = (f"Arguments {intersection} are deprecated, whether passed "
+                           "by position or keyword. They will be removed in SciPy "
+                           "{version}. ")
+                message += custom_message
+                warnings.warn(message, category=DeprecationWarning, stacklevel=3)
+
         @functools.wraps(f)
         def inner_f(*args, **kwargs):
+
             extra_args = len(args) - len(all_args)
             if extra_args <= 0:
+                warn_deprecated_args(kwargs)
                 return f(*args, **kwargs)
 
             # extra_args > 0
-            args_msg = ", ".join(kwonly_args[:extra_args])
+            kwonly_extra_args = set(kwonly_args[:extra_args]) - deprecated_args
+            args_msg = ", ".join(kwonly_extra_args)
```

Comment:
This warns about the deprecated *positional* use of arguments passed by position. We don't want the `deprecated_args` (which are completely decorated) to be included in that list; warn about those separately.